### PR TITLE
[CDAP-1554] Moved constants for default and system namespaces to Id.N…

### PR DESCRIPTION
**Pull request by bdmogal**
…amespace.

Also moved Constants.DEFAULT_NAMESPACE_META to NamespaceMeta.DEFAULT.
Other random cleanup along the way.

Jira: [CDAP-1554](https://issues.cask.co/browse/CDAP-1554)
Build: http://builds.cask.co/browse/CDAP-DUT2467

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/InstanceURIParser.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public CLIConnectionConfig parse(String uriString) {
     URI uri = URI.create(uriString);
     Id.Namespace namespace =
       (uri.getPath() == null || uri.getPath().isEmpty() || "/".equals(uri.getPath())) ?
-      Constants.DEFAULT_NAMESPACE_ID :
+      Id.Namespace.DEFAULT :
       Id.Namespace.from(uri.getPath().substring(1));
```
Comment (jayfjin) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/InstanceURIParser.java]: move this to the above line

File: cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java [hunk 2]
```diff
@@ -33,22 +33,21 @@ public void testParse() {
...
     int defaultPort = cConf.getInt(Constants.Router.ROUTER_PORT);
-    Id.Namespace defaultNamespace = Id.Namespace.from(Constants.DEFAULT_NAMESPACE);
     Id.Namespace someNamespace = Id.Namespace.from("nsx");
     InstanceURIParser parser = new InstanceURIParser(cConf);
 
-    Assert.assertEquals(new CLIConnectionConfig(defaultNamespace, "somehost", defaultPort, false),
+    Assert.assertEquals(new CLIConnectionConfig(Id.Namespace.DEFAULT, "somehost", defaultPort, false),
                         parser.parse("somehost"));
-    Assert.assertEquals(new CLIConnectionConfig(defaultNamespace, "somehost", defaultPort, false),
+    Assert.assertEquals(new CLIConnectionConfig(Id.Namespace.DEFAULT, "somehost", defaultPort, false),
```
Comment (jayfjin) [inline cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java]: same object as line 39?

File: cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java [hunk 3]
```diff
@@ -33,22 +33,21 @@ public void testParse() {
...
                         parser.parse("http://somehost"));
-    Assert.assertEquals(new CLIConnectionConfig(defaultNamespace, "somehost", defaultSSLPort, true),
+    Assert.assertEquals(new CLIConnectionConfig(Id.Namespace.DEFAULT, "somehost", defaultSSLPort, true),
                         parser.parse("https://somehost"));
 
-    Assert.assertEquals(new CLIConnectionConfig(defaultNamespace, "somehost", 1234, false),
+    Assert.assertEquals(new CLIConnectionConfig(Id.Namespace.DEFAULT, "somehost", 1234, false),
                         parser.parse("somehost:1234"));
-    Assert.assertEquals(new CLIConnectionConfig(defaultNamespace, "somehost", 1234, false),
+    Assert.assertEquals(new CLIConnectionConfig(Id.Namespace.DEFAULT, "somehost", 1234, false),
```
Comment (jayfjin) [inline cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java]: same object as line 46?

**Comment by jayfjin** [comment]
minor nits, otherwise LGTM

File: cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java (hunk 2 above)
Comment (bdmogal) [inline cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java]: I think this was so all the assertions in this test align, and so are easy on the eyes for comparison, etc. Just moving out a couple as variables makes it a little hard to read.

File: cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java (hunk 3 above)
Comment (bdmogal) [inline cdap-cli/src/test/java/co/cask/cdap/cli/util/InstanceURIParserTest.java]: same as above

**Comment by bdmogal** [comment]
Build has passed - http://builds.cask.co/browse/CDAP-DUT2467-2. Merging.

**Pull request merged**